ui/widget.js

```diff
@@ -367,6 +371,20 @@ $.Widget.prototype = {
 		return this;
 	},
 	_setOption: function( key, value ) {
+		var classKey, elements;
+		if ( key === "classes" ) {
+			for ( classKey in value ) {
+				if ( value[ classKey ] !== this.options.classes[ classKey ] ) {
+					if ( this.classObject[ classKey ] && this.classObject[ classKey ].length > 0 ) {
+						elements =  $.extend( {}, this.classObject[ classKey ] );
+						this._removeClass( this.classObject[ classKey ], classKey );
+						elements.addClass(
+							this._classes( elements, classKey, null, true, value ) );
```

Comment:
I assume you mean `._addClass` and its because _addClass always omits the last argument to `_classes` and so it uses  `this.options`, here we are passing the value for classes which was passed to `_setOption` to classes so that it uses that instead.

Note: I already switched `_classes` to use a settings object instead of 5 arguments as @jzaefferer suggested.